File: 0049-group-anagrams/0049-group-anagrams.cs
public class Solution {
    public IList<IList<string>> GroupAnagrams(string[] strs) {
        var ret = new List<IList<string>>();
        var dic = new Dictionary<string, List<string>>();
        
        foreach(var s in strs)
        {
            var chars = s.ToCharArray();
            Array.Sort(chars);
            string keyStr = new String(chars);
            if(dic.ContainsKey(keyStr))
                dic[keyStr].Add(s);
            else {
                dic[keyStr] = new List<string>() {s};
            }
        }
        
        foreach(var d in dic){
            ret.Add(d.Value);
        }
        
        return ret;
    }
}

File: 0049-group-anagrams/0049-group-anagrams.ts
function groupAnagrams(strs: string[]): string[][] {
    const hm: Map<string, string[]> = new Map();
    const ret: string[][] = [];
    for(const s of strs){
        //const charArray = [...s];
        //charArray.sort();
        //let keyStr = charArray.toString();
        const keyStr = s.split('').sort().join('');
        
        if(!hm.has(keyStr)) hm.set(keyStr, []);
        hm.get(keyStr).push(s);
    }
    
    return [...hm.values()];
};